src/main/java/com/kkm/kkm_server_v2/domain/post/service/DistanceAndCategoryPostService.java

```diff
@@ -0,0 +1,47 @@
+package com.kkm.kkm_server_v2.domain.post.service;
+
+import com.kkm.kkm_server_v2.domain.location.domain.Location;
+import com.kkm.kkm_server_v2.domain.location.facade.LocationFacade;
+import com.kkm.kkm_server_v2.domain.post.domain.Post;
+import com.kkm.kkm_server_v2.domain.post.domain.enums.PostCategory;
+import com.kkm.kkm_server_v2.domain.post.domain.repository.PostRepository;
+import com.kkm.kkm_server_v2.domain.post.presentation.dto.response.PostListResponse;
+import com.kkm.kkm_server_v2.domain.post.presentation.dto.response.PostResponse;
+import com.kkm.kkm_server_v2.domain.user.domain.User;
+import com.kkm.kkm_server_v2.domain.user.facade.UserFacade;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class DistanceAndCategoryPostService {
+    private final PostRepository postRepository;
+    private final UserFacade userFacade;
+    private final LocationFacade locationFacade;
+
+    @Transactional
+    public PostListResponse execute(int page, int size, PostCategory category, int distance) {
+        User user = userFacade.getCurrentUser(false);
+        Location location = locationFacade.findBySelectedAndUserLocation(user);
+        Pageable pageable = PageRequest.of(page - 1, size, Sort.Direction.DESC, "pullDate");
+        Page<Post> list;
+        if (category == PostCategory.ALL) {
+            list = postRepository.findByDistance(location.getLongitude(), location.getLatitude(), distance, pageable);
+        } else {
+            list = postRepository.findByDistanceAndCategory(location.getLongitude(), location.getLatitude(), distance, category, pageable);
+        }
```

Comment:
if 문 말고 삼항 연산자로 사용하는 거는 어떤가요? 별로면 말고 ㅎㅎ
```suggestion
        Page<Post> list = category == PostCategory.ALL ? postRepository.findByDistance(location.getLongitude(), location.getLatitude(), distance, pageable) : postRepository.findByDistanceAndCategory(location.getLongitude(), location.getLatitude(), distance, category, pageable);
```

Reply:
좋은 생각인 것 같아요 코드가 더 간결해지네용